### PR TITLE
Computer vision framework

### PR DESCRIPTION
**Pull request by dewagter**
instead of having many new copies of the same, a computer vision framework is initiated:

video_thread

which can run one or more computer vision modules (e.g. rtp_stream or qrcode or opticflow)

### CODE REVIEW

File: sw/airborne/modules/computer_vision/cv.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * @file modules/computer_vision/cv.h
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/cv.c]: plz fix file path
also please make use two spaces indent

**Comment by flixr** [comment]
I guess the viewvideo_nps.c could be removed (it doesn't do anything anyway) and the modules disabled for the nps target...

**Comment by dewagter** [comment]
I would like to implement viewvideo_nps instead of removing it.

**Comment by flixr** [comment]
even better, although I wonder a bit what the random image will be good for ;-)

**Comment by podhrmic** [comment]
Maybe you can somehow link it with flight gear and get some kind of video from the "cabin"?

**Comment by flixr** [comment]
Looks good to me, without having had the chance to actually test it....

**Comment by dewagter** [comment]
@flixr obviously the test image is not very useful indeed, but allows checking of the computer vision routines in simulation. Note that the video_thread code from AP can also run in linux, showing your webcam image while simulating. Not sure how this should be done when you have 3 options for the simulation code: 1) (no vision in nps) 2) video_thread_nps (test image or flightgear/ROS/smartuav coupling), 3) webcam.

**Pull request merged**